File: spark_java/src/main/java/com/spark/sql/study/dataset/ManuallySpecifyOptions.java
package com.spark.sql.study.dataset;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SQLContext;

/**
 * @author yangqian
 * @date 2021/5/12
 */
public class ManuallySpecifyOptions {

    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setAppName("ManuallySpecifyOptions").setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);
        SQLContext sqlContext = new SQLContext(sc);

        Dataset<Row> peopleDS = sqlContext
                .read()
                .format("json")
                .load("/Users/qian/WorkSpaces/own-workspace/2021/spark_code_repository/spark_java/src/main/resources/spark_sql/people.json");

        peopleDS.select("name")
                .show();

        peopleDS.select("name")
                .write().format("parquet")
                .save("/Users/qian/WorkSpaces/own-workspace/2021/spark_code_repository/spark_java/src/main/resources/spark_sql/peopleName_java.parquet");

    }

}

File: spark_scala/src/main/scala/com/spark/sql/scala/lesson/dataset/ManuallySpecifyOptions.scala
package com.spark.sql.scala.lesson.dataset

import org.apache.spark.sql.SparkSession
import org.apache.spark.{SparkConf, SparkContext}

/**
 * @date 2021/5/12
 */
object ManuallySpecifyOptions {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("ManuallySpecifyOptions").setMaster("local")
    val sc = new SparkContext(conf)
    val sqlContext = SparkSession.builder().config(conf).getOrCreate()

    val peopleDS = sqlContext.read.format("json")
      .load("/Users/qian/WorkSpaces/own-workspace/2021/spark_code_repository/spark_java/src/main/resources/spark_sql/people.json")
    peopleDS.select("name")
      .write.format("parquet")
      .save("/Users/qian/WorkSpaces/own-workspace/2021/spark_code_repository/spark_java/src/main/resources/spark_sql/peopleName_scala.parquet")
  }

}
